tools/bmh_test/clean_local_bmh_test_setup.sh

```diff
@@ -1,24 +1,24 @@
 #!/usr/bin/env bash
 
-set -eux
+set -ux
 
+BMH_NAME_REGEX="${1:-^bmh-test-}"
 # Get a list of all virtual machines
-VM_LIST=$(virsh -c qemu:///system list --all --name | grep '^bmh-test-') || true
+VM_LIST=$(virsh -c qemu:///system list --all --name | grep "${BMH_NAME_REGEX}")
 
 if [[ -n "${VM_LIST}" ]]; then
     # Loop through the list and delete each virtual machine
     for vm_name in ${VM_LIST}; do
         virsh -c qemu:///system destroy --domain "${vm_name}"
         virsh -c qemu:///system undefine --domain "${vm_name}" --remove-all-storage
-        kubectl delete baremetalhost "${vm_name}" || true
+        kubectl delete baremetalhost "${vm_name}"
     done
 else
     echo "No virtual machines found. Skipping..."
 fi
 
 # Clear vbmc
-docker stop vbmc
```

Comment:
whats the harm if we stop it first, why are we removing this? Is there a possibility that it might not exist?

Reply:
The `-f` makes it unnecessary. We remove it no matter if it runs or not